cmd/job_receiver/main.go

```diff
@@ -36,6 +39,18 @@ func initRedis(cfg *config.Config) (*redis.Client, error) {
 	return client, nil
 }
 
+func verifyConfiguration(cfg *config.Config) error {
```

Comment:
I thought about that as well.  I think it is something we need to look into.  If it viper does support something like that, then it might make sense to break up the configuration into "gateway" options and "switch" options.  This way we can mark some options as required for the gateway, but not required for the switch, etc.  Of course, it would be nice to share the common configuration options so that we didn't duplicate the logic to support common config options.